modules/account-lib/test/unit/coin/algo/transactionBuilder/transferBuilder.ts

```diff
@@ -0,0 +1,102 @@
+import crypto from 'crypto';
+import { coins } from '@bitgo/statics';
+import should from 'should';
+import sinon, { assert } from 'sinon';
+import { AddressValidationError, TransferBuilder } from '../../../../../src/coin/algo';
+import * as AlgoResources from '../../../../resources/algo';
+
+describe('Algo Transfer Builder', () => {
+  let builder: TransferBuilder;
+
+  const sender = AlgoResources.accounts.account1;
+  const receiver = AlgoResources.accounts.account2;
+
+  beforeEach(() => {
+    const config = coins.get('algo');
+    builder = new TransferBuilder(config);
+  });
+
+  describe('setter validation', () => {
+    it('should validate receiver address is a valid algo address', () => {
+      const spy = sinon.spy(builder, 'to');
+      should.throws(
+        () => builder.to({ address: 'wrong-addr' }),
+        (e: Error) => e.name === AddressValidationError.name,
+      );
+      should.doesNotThrow(() => builder.to({ address: sender.address }));
+      assert.calledTwice(spy);
+    });
+
+    it('should validate transfer amount', () => {
+      const spy = sinon.spy(builder, 'amount');
+      should.throws(
+        () => builder.amount(-1),
+        (e: Error) => e.message === 'Value cannot be less than zero',
+      );
+      should.throws(
+        () => builder.amount(undefined),
+        (e: Error) => e.message === 'Amount should not be undefined',
+      );
+      should.doesNotThrow(() => builder.amount(1000));
+      assert.calledThrice(spy);
+    });
+  });
+
+  describe('build transfer transaction', () => {
+    it('should build a transfer transaction', async () => {
+      builder.sender({ address: sender.address });
```

Comment:
can chain these to make it more concise.

 builder.sender({ address: sender.address }).to({ address: receiver.address }).amount(10000)... etc